include/pistache/http.h

```diff
@@ -316,6 +316,14 @@ class ResponseStream : public Message {
     friend
     ResponseStream& operator<<(ResponseStream& stream, const T& val);
 
+    std::streamsize write(const char * data, std::streamsize sz) {
+        std::ostream os(&buf_);
+        os << std::hex << sz << crlf;
```

Comment:
How would one send raw binary data without the hex size prefix and crlf tail?  Maybe you should rename this method to "writeChunkedBinaryFormat()"?  To me "write()" implies that its not going to modify what is being written.

Reply:
If the encoding weren't chunked, the server could just send the binary data as the body, and it would be up to the client to give it meaning. I'm using this approach now, when gzip'ing a serialized object. Just writing the data into a std::string and sending it as the body of the response works just fine.

I don't object to changing the name, but the ResponseStream is explicitly returning encoded chunks. I didn't think it needed to be made explicit in the write method. If that's your only objection, though, I'm happy to do it.

Since the data passed into the function needn't be binary, it could be any arbitrary data, would `writeChunk()` ok?

Reply:
Sorry, I forgot to mention sending the raw binary data works because the content-length is specified, so the client knows how many bytes to expect, so it replaces the hex size and trailing crlf.

Reply:
I'm changing my mind. The hex size and the trailing crlf are (and should be) hidden implementation details of the stream. Just as `operator<<` doesn't advertise that it's mangling the output, the `write()` call is just responsible for getting the data to the client, which it does. I think that `write()` is the appropriate name.

There's also nice symmetry around the operator<< using the << operator of the buf, whereas the write() call uses the write() of the buf.

Reply:
I don't like that a "write" method modifies the data that's being written.  I think that it might lead to user confusion in the future.  However, I have no strong argument against your change either.  I will merge your PR after your next response.